### ISSUE TITLE
Add a field on Cluster Status indicating the control plane is ready

### ISSUE THREAD

**Issue by chuckha** [kubernetes-sigs/cluster-api#1243]
/kind feature

**Describe the solution you'd like**
CAPI uses an annotation to mark a cluster with a control plane ready for joining new nodes. Instead of using an annotation we should use a field on Status.

**Anything else you would like to add:**
See https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/issues/99

/cc @detiber @fabriziopandini @ykakarap

**Comment by ykakarap** [comment]
Since this is related to an issue I am working on in CABPK, I would like to take this up.

/assign @ykakarap

**Comment by vincepri** [comment]
@chuckha I don't think CAPI uses any annotations today to mark a control plane ready, were you referring to CAPA? 

@ykakarap Before starting implementing `Status.ControlPlaneReady` field, we should first update the cluster proposal document to reflect this change and include details such as: how will we determine if a Cluster's control plane is ready, how can we make sure we can re-calculate readiness given that the value is going to be in Status.

**Comment by chuckha** [comment]
@vincepri  https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/controller/machine/machine_controller_phases.go#L279

is what I'm referencing

**Comment by vincepri** [comment]
Things past me wrote and I forgot 😂

**Comment by moshloop** [comment]
This will need to consider https://github.com/kubernetes-sigs/cluster-api/issues/1197 and https://github.com/kubernetes-sigs/cluster-api/issues/1250  which could potentially influence this.

**Comment by ncdc** [comment]
I am in favor of making this change. We have it as an annotation in v1alpha1, mainly because we didn't want to modify the API. I think it makes sense to promote this to a regular field for v1alpha2 so it is more visible.

As far as #1197 and #1250, there is not time for either of those to land in v1alpha2, so (pending lazy consensus) this should be able to go in by itself.

**Comment by ykakarap** [comment]
I am guessing we only want to add the field for now and considering refactoring the core to use this field instead of the annotation in a follow-up issue, correct?

**Comment by ncdc** [comment]
I'd rather do it in a single PR. But we should open a PR to modify the proposal first.

I'd actually like to propose we consider naming it `ControlPlaneInitialized` instead of `ControlPlaneReady`. I don't think we necessarily need to constantly monitor the health of the control plane and update this bool. I'd rather use it as a gate that only ever changes from false to true a single time.
